Sources/SwiftDocC/Utility/FoundationExtensions/String+Capitalization.swift

```diff
@@ -19,23 +19,33 @@ extension String {
         
         guard !self.isEmpty else { return self }
         
-        let selfTrimmedWhitespace = self.trimmingCharacters(in: .whitespacesAndNewlines)
+        guard let firstWordStartIndex = self.firstIndex(where: { !$0.isWhitespace && !$0.isNewline }) else { return self }
         
-        let firstWord = selfTrimmedWhitespace.prefix(while: { !$0.isWhitespace && !$0.isNewline })
+        // Find where the first word starts: this is additional processing to handle white spaces before the first word.
+        let firstWord = self[firstWordStartIndex...].prefix(while: { !$0.isWhitespace && !$0.isNewline})
         
-        guard firstWord.count > 0 else { return self }
         
+        guard firstWord.count > 0 else { return self }
         let firstWordCharacters = CharacterSet.init(charactersIn: String(firstWord))
```

Comment:
Upon further though the fact that we are creating a temporary string here is not great either I came up with the following refactoring of the logic:
```swift
extension String {
    // Notice this addition precomputing the CharacterSet to use in the function
    private static let charactersPreventingWordCapitalization = 
        CharacterSet.lowercaseLetters.union(.punctuationCharacters).inverted
    
    func capitalizeFirstWord() -> String {
        guard let firstWordStartIndex = self.firstIndex(where: { !$0.isWhitespace && !$0.isNewline }) else { return self }
        let firstWord = self[firstWordStartIndex...].prefix(while: { !$0.isWhitespace && !$0.isNewline})
        
        // Notice the change of logic here which enables us to not create the temporary, 
        // additionally precomputing the character set to use saves time during the function.
        guard firstWord.rangeOfCharacter(from: Self.charactersPreventingWordCapitalization) == nil else {
            return self
        }
        
        var resultString = String()
        resultString.reserveCapacity(self.count)
        resultString.append(contentsOf: self[..<firstWordStartIndex])
        resultString.append(contentsOf: String(firstWord).localizedCapitalized)
        let restStartIndex = self.index(firstWordStartIndex, offsetBy: firstWord.count)
        resultString.append(contentsOf: self[restStartIndex...])
        return resultString
    }
}
```